### PR TITLE
Acceptance tests: enable expiration date for user and group

### PR DESCRIPTION
**Pull request by haribhandari07**
## Description

Adds acceptance tests for enabling expiration date as admin for user and group share.
This PR adds acceptance tests for PR https://github.com/owncloud/core/pull/36324
## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- CI
- Manual
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php [hunk 1]
```diff
@@ -240,6 +240,72 @@ public function theAdministratorEnablesExcludeGroupsFromSharingUsingTheWebui() {
 		);
 	}
 
+	/**
+	 * @When the administrator enables default expiration date for user shares
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php]:
```suggestion
	 * @When the administrator enables default expiration date for user shares using the webUI
```

File: tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php [hunk 2]
```diff
@@ -240,6 +240,72 @@ public function theAdministratorEnablesExcludeGroupsFromSharingUsingTheWebui() {
...
+	 * @return void
+	 */
+	public function administratorEnablesDefaultExpirationDateForUserShares() {
+		$this->adminSharingSettingsPage->enableDefaultExpirationDateForUserShares(
+			$this->getSession()
+		);
+	}
+
+	/**
+	 * @When the administrator enforces maximum expiration date for user shares
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php]:
```suggestion
	 * @When the administrator enforces maximum expiration date for user shares using the webUI
```

File: tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php [hunk 3]
```diff
@@ -240,6 +240,72 @@ public function theAdministratorEnablesExcludeGroupsFromSharingUsingTheWebui() {
...
+	 * @return void
+	 */
+	public function administratorEnforcesMaximumExpirationDateForUserShares() {
+		$this->adminSharingSettingsPage->enforceMaximumExpirationDateForUserShares(
+			$this->getSession()
+		);
+	}
+
+	/**
+	 * @When the administrator enforces maximum expiration date for group shares
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php]:
```suggestion
	 * @When the administrator enforces maximum expiration date for group shares using the webUI
```

File: tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php [hunk 4]
```diff
@@ -240,6 +240,72 @@ public function theAdministratorEnablesExcludeGroupsFromSharingUsingTheWebui() {
...
+	 * @return void
+	 */
+	public function administratorEnforcesMaximumExpirationDateForGroupShares() {
+		$this->adminSharingSettingsPage->enforceMaximumExpirationDateForGroupShares(
+			$this->getSession()
+		);
+	}
+
+	/**
+	 * @When the administrator updates the user share expiration date to :days days
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php]:
```suggestion
	 * @When the administrator updates the user share expiration date to :days days using the webUI
```

File: tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php [hunk 5]
```diff
@@ -240,6 +240,72 @@ public function theAdministratorEnablesExcludeGroupsFromSharingUsingTheWebui() {
...
+	 * @param int $days
+	 *
+	 * @return void
+	 */
+	public function administratorUpdatesUserShareExpirationTo($days) {
+		$this->adminSharingSettingsPage->setExpirationDaysForUserShare($days, $this->getSession());
+	}
+
+	/**
+	 * @When the administrator updates the group share expiration date to :days days
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php]:
```suggestion
	 * @When the administrator updates the group share expiration date to :days days using the webUI
```

File: tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php [hunk 6]
```diff
@@ -240,6 +240,72 @@ public function theAdministratorEnablesExcludeGroupsFromSharingUsingTheWebui() {
...
+	 * @param int $days
+	 *
+	 * @return void
+	 */
+	public function administratorUpdatesGroupShareExpirationTo($days) {
+		$this->adminSharingSettingsPage->setExpirationDaysForGroupShare($days, $this->getSession());
+	}
+
+	/**
+	 * @When the administrator enables default expiration date for group shares
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php]:
```suggestion
	 * @When the administrator enables default expiration date for group shares using the webUI
```

File: tests/acceptance/features/lib/AdminSharingSettingsPage.php [hunk 7]
```diff
@@ -339,6 +352,132 @@ public function enableExcludeGroupFromSharing(Session $session) {
...
+	 *
+	 * @throws ElementNotFoundException
+	 * @return NodeElement|NULL
+	 */
+	private function findGroupShareExpirationField() {
+		$expirationDateField = $this->find("xpath", $this->groupShareExpirationDateFieldXpath);
+		$this->assertElementNotNull(
+			$expirationDateField,
+			__METHOD__ .
+			" xpath $this->groupShareExpirationDateFieldXpath could not find set-user-share-expiration-field"
```
Comment (phil-davis) [inline tests/acceptance/features/lib/AdminSharingSettingsPage.php]:
```suggestion
			" xpath $this->groupShareExpirationDateFieldXpath could not find set-group-share-expiration-field"
```

**Comment by HariBhandari07** [comment]
@phil-davis I have added additional acceptance tests. Please have a look.

File: tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php [hunk 8]
```diff
@@ -98,6 +98,80 @@ public function adminTogglesShareViaLink($action) {
 		);
 	}
 
+	/**
+	 *
+	 * @param string $checkbox
+	 *
+	 * @return void
+	 */
+	public function checkCheckBoxIsChecked($checkbox) {
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php]: `assertCheckBoxIsChecked` would be a better name
function can be private

File: tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php [hunk 9]
```diff
@@ -98,6 +98,80 @@ public function adminTogglesShareViaLink($action) {
...
+		Assert::assertTrue($checkbox->isChecked(), "checkbox is not checked");
+	}
+
+	/**
+	 * @Then settings set default expiration date for user shares should be enabled on the webUI
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebUIAdminSharingSettingsContext.php]: `Then the default expiration date checkbox for user shares should be enabled on the webUI`

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/lib/AdminSharingSettingsPage.php [hunk 10]
```diff
@@ -338,6 +351,175 @@ public function enableExcludeGroupFromSharing(Session $session) {
 			$this->excludeGroupFromSharingCheckboxId
 		);
 	}
+	/**
+	 * @return string
```
Comment (individual-it) [inline tests/acceptance/features/lib/AdminSharingSettingsPage.php]:
```suggestion
	 * @return NodeElement|null
```
same for similar functions and functions that use these kind of functions

**Review by individual-it — CHANGES_REQUESTED** [review]

**Comment by individual-it** [comment]
cherry picked to and fixed the issues in https://github.com/owncloud/core/pull/36531/

**Pull request closed without merging**